### PR TITLE
CDPSDX-581: Implement service API's to delete service principal and host

### PR DESCRIPTION
**Pull request by kkalvagadda**
Added API's to delete host and service.

### CODE REVIEW

File: freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java [hunk 1]
```diff
@@ -108,14 +109,29 @@ public ServiceKeytabResponse getExistingServiceKeytab(ServiceKeytabRequest reque
...
+
+        Map<String, String> principalCanonicalMap = ipaClient.findAllService().stream()
+                .collect(Collectors.toMap(com.sequenceiq.freeipa.client.model.Service::getKrbprincipalname,
+                        com.sequenceiq.freeipa.client.model.Service::getKrbcanonicalname));
+        Set<String> services = principalCanonicalMap.entrySet().stream().filter(e -> e.getKey().contains(request.getServerHostName()))
+                .map(Map.Entry::getValue).collect(Collectors.toSet());
+        for (String service : services) {
+            ipaClient.deleteService(service);
+        }
+        ipaClient.deleteHost(request.getServerHostName());
```
[the remark this answers is not in the document]
Comment (kkalvagadda) [reply freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java]: As the documentation doesn't that say that deleting the host would clean up all the services associated with that service, I think it is safe to delete the services explicitly.

File: freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java [hunk 2]
```diff
@@ -108,14 +109,29 @@ public ServiceKeytabResponse getExistingServiceKeytab(ServiceKeytabRequest reque
...
-        return;
+    public void deleteHost(HostRequest request, String accountId) throws FreeIpaClientException {
+        FreeIpaClient ipaClient;
+        Stack freeIpaStack = getFreeIpaStack(request.getEnvironmentCrn(), accountId);
+        ipaClient = freeIpaClientFactory.getFreeIpaClientForStack(freeIpaStack);
+
+        Map<String, String> principalCanonicalMap = ipaClient.findAllService().stream()
+                .collect(Collectors.toMap(com.sequenceiq.freeipa.client.model.Service::getKrbprincipalname,
+                        com.sequenceiq.freeipa.client.model.Service::getKrbcanonicalname));
+        Set<String> services = principalCanonicalMap.entrySet().stream().filter(e -> e.getKey().contains(request.getServerHostName()))
```
Comment (lacikaaa) [inline freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java]: this 2 streams could be transformed into 1

File: freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java [hunk 3]
```diff
@@ -265,4 +281,8 @@ private String constructVaultPath(String accountId, String type, String subtype)
         return String.format("%s/%s/%s/%s-%s", accountId, type, subtype,
                 UUID.randomUUID().toString(), Long.toHexString(System.currentTimeMillis()));
     }
+
+    private static String constructCanonicalPrincipal(String serviceName, String hostName, String realm) {
```
Comment (lacikaaa) [inline freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java]: this duplicates the logic in `serviceAdd` method

Comment (kkalvagadda) [reply freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java]: I wrongly resolved the merge conflict where rebasing the patch. Will address it.

Comment (kkalvagadda) [reply freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java]: Addressed.

**Pull request merged**